src/pudl/transform/sec10k.py

```diff
@@ -0,0 +1,471 @@
+            "former_conformed_name": "company_name_old",
+            "location_of_inc": "location_of_incorporation",
+            "state_of_incorporation": "incorporation_state",
+            "irs_number": "taxpayer_id_irs",
+            "parent_company_cik": "parent_company_central_index_key",
+            "files_10k": "files_sec10k",
+            "date_of_name_change": "name_change_date",
+        }
+    ).assign(
+        filename_sec10k=lambda x: _simplify_filename_sec10k(x["filename_sec10k"]),
+        fraction_owned=lambda x: _compute_fraction_owned(x["ownership_percentage"]),
+    )
+
+    df[["industry_name_sic", "industry_id_sic"]] = (
+        _standardize_industrial_classification(df["standard_industrial_classification"])
+    )
+    df["taxpayer_id_irs"] = _standardize_taxpayer_id_irs(df["taxpayer_id_irs"])
+
+    # Some utilities harvested from EIA 861 data that don't show up in our entity
+    # tables. These didn't end up improving coverage, and so will be removed upstream.
+    # Hack for now is to just drop them so the FK constraint is respected.
+    # See https://github.com/catalyst-cooperative/pudl/issues/4050
+    bad_utility_ids = [
+        3579,  # Cirro Group, Inc. in Texas
+    ]
+    df = df[~df.utility_id_eia.isin(bad_utility_ids)]
+
+    return df
+
+
+@dg.asset(
+    io_manager_key="pudl_io_manager",
+    group_name="core_sec10k",
+)
+def core_sec10k__quarterly_exhibit_21_company_ownership(
+    raw_sec10k__exhibit_21_company_ownership: pd.DataFrame,
+) -> pd.DataFrame:
+    """Standardize the contents of the SEC 10-K exhibit 21 company ownership table."""
+    df = raw_sec10k__exhibit_21_company_ownership.rename(
+        columns={
+            "sec10k_filename": "filename_sec10k",
+            "subsidiary": "subsidiary_company_name",
+            "location": "subsidiary_company_location",
+        }
+    ).assign(
+        filename_sec10k=lambda x: _simplify_filename_sec10k(x["filename_sec10k"]),
+        fraction_owned=lambda x: _compute_fraction_owned(x["ownership_percentage"]),
+        report_date=lambda x: _year_quarter_to_date(x["year_quarter"]),
+    )
+    return df
+
+
+@dg.asset(
+    io_manager_key="pudl_io_manager",
+    group_name="core_sec10k",
+)
+def core_sec10k__assn_sec10k_filers_and_eia_utilities(
+    core_sec10k__parents_and_subsidiaries: pd.DataFrame,
+) -> pd.DataFrame:
+    """An association table between SEC 10k filing companies and EIA utilities."""
+    sec_eia_matches = (
+        core_sec10k__parents_and_subsidiaries.loc[
+            :, ["central_index_key", "utility_id_eia"]
+        ]
+        .drop_duplicates(subset="central_index_key")
+        .dropna()
+    )
+    # TODO: this assertion is failing, so I added the drop_duplicates() back in above,
+    # but should ask Katie about why and if it's expected / important...
+    # Verify that each CIK is matched to only one utility:
+    # assert sec_eia_matches["central_index_key"].nunique() == len(sec_eia_matches)
```

Comment:
@katie-lamb left a comment in the previous version of this asset that indicated each CIK should only appear once in this table, but that expectation doesn't seem to be met. Do we know why? Is it a problem?

Reply:
The previous comment was a little misleading -- duplicates need to be dropped because we're reconstructing an association table from an un-normalized table, which has many records which refer to the same companies.  In addition to that, most of the records have NA values in one or both of the `central_index_key` and `utility_id_eia` columns, and those were also duplicates.

I've clarified the expectations and am asserting that both columns are independently unique. This could probably go into a `dbt` test instead.  They're already required to be jointly unique and non-null because they are the PK for the table.